lib/src/configuration.cc

```diff
@@ -683,6 +683,22 @@ void Configuration::validateAndNormalizeOtherSettings() {
         }
 
         auto pid_dir = fs::path(pid_file).parent_path().string();
+
+#ifdef __sun
+        // Solaris cleans up /var/run when rebooting; daemonization
+        // assumes that the PID file directory exists. If we're using
+        // the default directory, make sure it exists (PCP-297)
+        if (pid_dir == "/var/run/puppetlabs" && !fs::exists(pid_dir)) {
```

Comment:
Yeah, on sleeping on it, I agree with restricting this to just when `pid_dir == "/var/run/puppetlabs"`.

I still like the idea about having this fix in place for non-solaris os's or installs which mount /var/run on a temp filesystem (i.e. my comment above, not this comment :))